### PR TITLE
Docs 13883 add rand number generator v5.0

### PR DESCRIPTION
**Pull request by davemungo**

### CODE REVIEW

File: source/release-notes/5.0.txt
Comment (andf-mongodb) [inline source/release-notes/5.0.txt:59]: Extra space here.

File: source/reference/operator/query.txt
Comment (andf-mongodb) [inline source/reference/operator/query.txt:357]: Why change the $comment definition on your $rand ticket? This is not what $comment does ...

File: source/release-notes/5.0.txt
Comment (andf-mongodb) [inline source/release-notes/5.0.txt:61]: mechanism --> method

File: source/reference/operator/query-others.txt
Comment (andf-mongodb) [inline source/reference/operator/query-others.txt:25]: Why change the $comment definition on your $rand ticket? This is not what $comment does ...

**Review by andf-mongodb — CHANGES_REQUESTED** [review]
Some small fixups, and a question regarding your change to $comment's definition.

File: source/reference/operator/query.txt
Comment (davemungo) [reply source/reference/operator/query.txt:357]: I didn't make that change on purpos. I must have gotten picked up during an update somehow.

File: source/release-notes/5.0.txt
Comment (davemungo) [reply source/release-notes/5.0.txt:59]: Done

File: source/reference/operator/query.txt
Comment (davemungo) [reply source/reference/operator/query.txt:357]: Current version looks like this on those lines. There's no mention of $comment: 

   * - :projection:`$meta`

     - Projects the document's score assigned during :query:`$text` operation.

   * - :projection:`$slice`

     - Limits the number of elements projected from an array. Supports skip and limit slices.

I'll put back the version with $slice and $without $comment.

Comment (davemungo) [reply source/reference/operator/query.txt:357]: Actually it's a lot more messed up than that. There's a bunch of stuff to fix.

**Comment by davemungo** [comment]
Everything should be in order now. The index pages have been updated and consistent

File: source/reference/operator/query.txt
Comment (andf-mongodb) [inline source/reference/operator/query.txt:352]: Can you bump the column width slightly so that the description field doesn't wrap by one character please?

File: source/reference/operator/query-miscellaneous.txt
Comment (andf-mongodb) [inline source/reference/operator/query-miscellaneous.txt:17]: Can you bump the column width slightly so that the description field doesn't wrap by one character please?

File: source/release-notes/5.0.txt
Comment (andf-mongodb) [inline source/release-notes/5.0.txt:61]: :pipeline:`$sampleRate` should be :expression:`$sampleRate`, no? Please also change this in the $sampleRate header just above. I think this is maybe more of DOCS-14053?

File: source/reference/operator/query/rand.txt
Comment (andf-mongodb) [inline source/reference/operator/query/rand.txt:18]: Please use the same syntax code block as you do in source/reference/operator/aggregation/rand.txt

i.e.:

:query:`$rand` has the following syntax:

   .. code-block:: javascript

      { $rand: {} }

**Review by andf-mongodb — CHANGES_REQUESTED** [review]
Some minor corrections when you get a chance, please.

File: source/reference/operator/query.txt
Comment (davemungo) [reply source/reference/operator/query.txt:353]: This is the problem we discussed before the break - sphynx is modifying the column widths. 

The settings here are the same as for the table above this one, however while that list-table fills the width of the page, this one only partially fills it. (See the 'Element' table above, it has the same problem. The same error appears on other index pages too.) 

None of these changes update the widths: 
-  Removing the " :class: singlecol " directive from line 11 
-  Changing  from 30, 70 to :widths: 25, 75
-  Adding a " :class: singlecol " directive to the list table
-  Various combinations of these

File: source/reference/operator/query/rand.txt
Comment (davemungo) [reply source/reference/operator/query/rand.txt:18]: Done. 

The syntax styles are very different both within and across the query and aggregation groups. Perhaps we need to agree on a standard form.

File: source/release-notes/5.0.txt
Comment (davemungo) [reply source/release-notes/5.0.txt:61]: Done

File: source/reference/operator/query-miscellaneous.txt
Comment (davemungo) [reply source/reference/operator/query-miscellaneous.txt:17]: See above

**Comment by davemungo** [comment]
I think this is ready now. 

Staging links: 
# Content Links:
https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/aggregation/rand.html#exp._S_rand

https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/query/rand.html#op._S_rand

# Index Pages 
https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/query.html

https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/query-miscellaneous.html

https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/aggregation.html#miscellaneous-operators

# Release notes
https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/release-notes/5.0.html#rand

File: source/release-notes/5.0.txt
Comment (davemungo) [reply source/release-notes/5.0.txt:59]: Done

File: source/release-notes/5.0.txt
Comment (davemungo) [reply source/release-notes/5.0.txt:51]: Done

**Comment by jeff-allen-mongo** [comment]
@davemungo, I noticed one more thing which is a must-fix here. Your staged query operator page for $rand has bad syntax, leading the opening word to render as `:query:$rand`, instead of `$rand`. I believe you want

```
query:`$rand`
```

Currently, you're missing the backticks.

Also (lower priority, but still would be nice), on the Aggregation page for $rand, we sometimes call it an "operator" and sometimes call it an "expression". Since it's nested under "Aggregation Pipeline Operators" can we please update that copy to consistently refer to the "$rand operator" rather than "$rand expression"?

Thank you. Once these issues are addressed we can merge this in.

**Comment by davemungo** [comment]
I've fixed the backticks and change 'expression' to 'operator' on the aggregation page. Here are the staging links: 


https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/query/rand.html#op._S_rand

https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/aggregation/rand.html#exp._S_rand

File: source/reference/operator/aggregation/rand.txt
Comment (jeff-allen-mongo) [inline source/reference/operator/aggregation/rand.txt:20]: [minor] All other aggregation operator pages lead with the verb, rather than using the operator name. Therefore, I suggest we remove "The :expression:`$rand` operator" from this sentence. And have this sentence read:

"Returns a random float between 0 and 1."

For a couple examples, notice the wording of the introductions of:

- https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/aggregation/sampleRate.html#exp._S_sampleRate

- https://docs-mongodbcom-staging.corp.mongodb.com/docs/docsworker-xlarge/DOCS-13883-add-rand-number-generator-v5.0/reference/operator/aggregation/setEquals.html

File: source/reference/operator/query/rand.txt
Comment (jeff-allen-mongo) [inline source/reference/operator/query/rand.txt:18]: I would suggest reformatting slightly to  have the definition and then the syntax. That approach makes more sense to me as a reader.

```
:query:`$rand` returns a random float between 0 and 1. 

:query:`$rand` has the following syntax:

.. code-block:: javascript 

   { $rand: {} }
```

The formatting throughout our query operators is inconsistent, as you mentioned @davemungo. This is definitely something we should revisit as a whole and try to agree on a format to use throughout our operators.

**Review by jeff-allen-mongo — APPROVED** [review]
Approving with 2 small last thoughts. Will leave open since @andf-mongodb also requested changes and would like to have him formally approve before merging.

File: source/reference/operator/query/rand.txt
Comment (davemungo) [reply source/reference/operator/query/rand.txt:18]: Done

File: source/reference/operator/aggregation/rand.txt
Comment (davemungo) [reply source/reference/operator/aggregation/rand.txt:20]: Done

**Review by andf-mongodb — APPROVED** [review]
LGTM

**Pull request merged**
